apis/v1/httproute_types.go

```diff
@@ -612,9 +612,14 @@ type HTTPHeaderMatch struct {
 	Name HTTPHeaderName `json:"name"`
 
 	// Value is the value of HTTP Header to be matched.
+	// <gateway:experimental:description>
+	// Must consist of printable US-ASCII characters, optionally separated
+	// by single tabs or spaces. See: https://tools.ietf.org/html/rfc7230#section-3.2
+	// </gateway:experimental:description>
 	//
 	// +kubebuilder:validation:MinLength=1
 	// +kubebuilder:validation:MaxLength=4096
+	// <gateway:experimental:validation:Pattern=`^[!-~]+([\t ]?[!-~]+)*$`>
```

Comment:
Can you add some invalid and valid examples to https://github.com/kubernetes-sigs/gateway-api/tree/main/hack/invalid-examples/experimental and https://github.com/kubernetes-sigs/gateway-api/tree/main/examples/experimental respectively to confirm this is working as intended?

Reply:
One note - this is a header value _matcher_ that supports two types of matches - exact and regex. I believe all forms of regex matcher used here would also be contained within US-ASCII characters, but would appreciate another confirmation of that.

Reply:
That's a valid point. AFAIK, non-printable characters aren't used as meta characters in regex patterns.